credoai/lens.py

```diff
@@ -348,36 +347,87 @@ class CredoData:
     -------------
     name : str
         Label of the dataset
-    X : data for model-input
-        Features passed to a model. Should be dataframe-like in the case of tabular data
-    y: data analogous model-output
-        Ground-truth labels
-    sensitive_features: array-like, optional
-        Array of sensitive-feature labels. E.g., protected attributes like race or gender
-        or categorical labels for important performance dimensions (["bright_lighting", "dark_lighting"])
+    data : pd.DataFrame
+        Dataset dataframe that includes all features and labels
+    sensitive_feature_key : str
+        Name of the sensitive feature column, like 'race' or 'gender'
+    label_key : str
+        Name of the label column
+    categorical_features_keys : list[str], optional
+        Names of the categorical features
+    omit_for_model_features_keys : list[str], optional
+        Names of the features to not use for model training
+        Include all the features in the data that were not used during model training
     metadata: dict, optional
         Arbitrary additional data that will be associated with the dataset
     """
-    name: str
-    X: "model-input"
-    y: "model-output"
-    sensitive_features: 'array-like' = None
-    metadata: dict = None
+    # name: str
+    # data: "model-input"
+    # sensitive_feature_key: "sensitive-feature-key"
+    # label_key: "model-output-key"
+    # categorical_features_keys: "array-like" = None
+    # omit_for_model_features_keys: "array-like" = None
+    # metadata: dict = None
+
+    def __init__(self,
+            name: str,
+            data: pd.DataFrame,
+            sensitive_feature_key: str,
+            label_key: str,
+            categorical_features_keys: Optional[List[str]]=None,
+            omit_for_model_features_keys: Optional[List[str]]=None): 
+
+        
+        self.data = data
+        self.sensitive_feature_key = sensitive_feature_key
+        self.label_key = label_key
+        self.categorical_features_keys = categorical_features_keys
+
+        self.name = name
+        self.sensitive_features = data[sensitive_feature_key]
+        self.y = data[label_key]
+        self.X = data.drop(columns=[label_key], axis=1)
+
+        if omit_for_model_features_keys:
+            self.X = self.X.drop(columns=omit_for_model_features_keys, axis=1, errors='ignore')
```

Comment:
Renamed it to `unused_features_keys`
I don't think so. If a user has used the sensitive feature for training, then it wouldn't be unused, which means they would leave `unused_features_keys` just as is, i.e. `None`.

Reply:
The primary use case of this assessment is to evaluate the fairness characteristics of the dataset. Almost all of the time the team _will_ remove the explicit sensitive feature. I feel this is optimizing for an edge case rather than the majority use. The end result is that almost every time the user will have to list that the sensitive feature should not be included in X. At the very least I'd like a default argument that adds the sensitive feature to `unused_feature_keys`. We can talk during standup tomorrow.